catalog/app/containers/Bucket/Queries/requests/athena.ts

```diff
@@ -0,0 +1,328 @@
+  try {
+    const executionIdsOutput = await athena
+      .listQueryExecutions({ WorkGroup: workgroup, NextToken: prev?.next })
+      .promise()
+
+    const ids = executionIdsOutput.QueryExecutionIds
+    if (!ids || !ids.length)
+      return {
+        list: [],
+        next: executionIdsOutput.NextToken,
+      }
+
+    const executionsOutput = await athena
+      ?.batchGetQueryExecution({ QueryExecutionIds: ids })
+      .promise()
+    const parsed = (executionsOutput.QueryExecutions || []).map(parseQueryExecution)
+    const list = (prev?.list || []).concat(parsed)
+    return {
+      list,
+      next: executionIdsOutput.NextToken,
+    }
+  } catch (e) {
+    // eslint-disable-next-line no-console
+    console.log('Unable to fetch')
+    // eslint-disable-next-line no-console
+    console.error(e)
+    throw e
+  }
+}
+
+export function useQueryExecutions(
+  workgroup: string,
+  prev: QueryExecutionsResponse | null,
+): AsyncData<QueryExecutionsResponse> {
+  const athena = AWS.Athena.use()
+  return useData(
+    fetchQueryExecutions,
+    { athena, prev, workgroup },
+    { noAutoFetch: !workgroup },
+  )
+}
+
+async function waitForQueryStatus(
+  athena: Athena,
+  QueryExecutionId: string,
+): Promise<Athena.QueryExecution | null> {
+  const statusData = await athena.getQueryExecution({ QueryExecutionId }).promise()
+  const status = statusData?.QueryExecution?.Status?.State
+  if (status === 'FAILED' || status === 'CANCELLED') {
+    throw new Error(status)
+  }
+
+  if (status === 'SUCCEEDED') {
+    return statusData?.QueryExecution || null
+  }
+
+  return waitForQueryStatus(athena, QueryExecutionId)
+}
+
+export interface QueryResultsResponse {
+  list: Athena.RowList
+  next?: string
+  queryExecution: QueryExecution | null
+}
+
+interface QueryResultsArgs {
+  athena: Athena
+  queryExecutionId: string
+  prev: QueryResultsResponse | null
+}
+
+async function fetchQueryResults({
+  athena,
+  queryExecutionId,
+  prev,
+}: QueryResultsArgs): Promise<QueryResultsResponse> {
+  const queryExecution = await waitForQueryStatus(athena, queryExecutionId)
+
+  const queryResultsOutput = await athena
+    .getQueryResults({
+      QueryExecutionId: queryExecutionId,
+      NextToken: prev?.next,
+    })
+    .promise()
+  const parsed = queryResultsOutput.ResultSet?.Rows || []
+  const list = (prev?.list || []).concat(parsed)
+  return {
+    list,
+    next: queryResultsOutput.NextToken,
+    queryExecution: queryExecution ? parseQueryExecution(queryExecution) : null,
+  }
+}
+
+export function useQueryResults(
+  queryExecutionId: string | null,
+  prev: QueryResultsResponse | null,
+): AsyncData<QueryResultsResponse> {
+  const athena = AWS.Athena.use()
+  return useData(
+    fetchQueryResults,
+    { athena, prev, queryExecutionId },
+    { noAutoFetch: !queryExecutionId },
+  )
+}
+
+async function hashQueryBody(queryBody: string, workgroup: string): Promise<string> {
```

Comment:
it could make sense to normalize query body (say, replace all whitespace with a single space, trim the string, maybe something else)